rental.java

```diff
@@ -0,0 +1,142 @@
+//package usaco2018january;
+//1 hour 45 min
+//1,2,7 is a success, 3,4 are wrong, 5,6,8,9,10 timed out
+//Difficulty 9/10
+//I thought this problem was really challenging, it made me think a lot and I felt like I may have took the wrong path in solving this problem

```

Comment:
This one is a bit messy. For messy code, it is super important to keep your top-level logic super clear! So you need add clear comments on your strategy.